packages/@glimmer/syntax/lib/syntax-error.ts

```diff
@@ -2,36 +2,20 @@ import { SourceSpan } from './source/span';
 
 export interface GlimmerSyntaxError extends Error {
   location: SourceSpan | null;
-  constructor: SyntaxErrorConstructor;
 }
 
-export interface SyntaxErrorConstructor {
-  new (message: string, location: SourceSpan | null): GlimmerSyntaxError;
-  readonly prototype: GlimmerSyntaxError;
-}
-
-/**
- * Subclass of `Error` with additional information
- * about location of incorrect markup.
- */
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export const GlimmerSyntaxError: SyntaxErrorConstructor = (function () {
-  SyntaxError.prototype = Object.create(Error.prototype);
-  SyntaxError.prototype.constructor = SyntaxError;
+export function generateSyntaxError(message: string, location: SourceSpan): GlimmerSyntaxError {
+  let { module, loc } = location;
+  let { line, column } = loc.start;
 
-  function SyntaxError(this: GlimmerSyntaxError, message: string, location: SourceSpan) {
-    let error = Error.call(this, message);
+  let code = location.asString();
+  let quotedCode = code ? `\n\n|\n|  ${code.split('\n').join('\n|  ')}\n|\n\n` : '';
 
-    let { module, loc } = location;
-    let { line, column } = loc.start;
+  let error = new Error(
+    `Syntax Error: ${message}: ${quotedCode}(error occurred in '${module}' @ line ${line} : column ${column})`
+  ) as GlimmerSyntaxError;
 
-    let code = location.asString();
-    let quotedCode = code ? `\n\n|\n|  ${code.split('\n').join('\n|  ')}\n|\n\n` : '';
+  error.location = location;
```

Comment:
You should set error.name = "SyntaxError" and don't include `Syntax Error:` in the message